### PR TITLE
Fixes #14335: Add sort, join and uniq

### LINKED ISSUE #14335: Add `uniq`, `join` and `sort` simple shell commands
For better scripting inside radare2 shell:
```sh
$ cat authors.txt
Chaucer
Chaucer
Orwell
Larkin
Larkin
$ uniq authors.txt
Chaucer
Orwell
Larkin
```

```sh
$ cat foodtypes.txt
1 Protein
2 Carbohydrate
3 Fat

$ cat foods.txt
1 Cheese 
2 Potato
3 Butter

$ join foodtypes foods.txt
1 Protein Cheese
2 Carbohydrate Potato
3 Fat Butter
```

### PR DESCRIPTION
**Pull request by deepakchethan**
After changes:

Screen Shot 2019-06-20 at 10 52 03 AM

### CODE REVIEW

**Comment by XVilka** [comment]
Please add a couple chapters in this section of radare2 book as well https://radare.gitbooks.io/radare2book/content/scripting/intro.html

File: libr/core/cmd_seek.c [hunk 1]
```diff
@@ -710,8 +738,16 @@ static int cmd_seek(void *data, const char *input) {
 		break;
 	}
 	case 'o': // "so"
-		cmd_seek_opcode (core, input + 1);
-		break;
+		switch (input[1]) {
+			case 'r':
```
Comment (radare) [inline libr/core/cmd_seek.c]: bad indentation

File: libr/core/cmd_seek.c [hunk 2]
```diff
@@ -710,8 +738,16 @@ static int cmd_seek(void *data, const char *input) {
...
+				if (input[2] == 't') {
+					cmd_sort (core, input);
```
Comment (radare) [inline libr/core/cmd_seek.c]: shouldnt this be input +2?

File: libr/core/cmd_seek.c [hunk 3]
```diff
@@ -710,8 +738,16 @@ static int cmd_seek(void *data, const char *input) {
...
+				}
```
Comment (radare) [inline libr/core/cmd_seek.c]: else invalid command?

File: libr/core/cmd_seek.c [hunk 4]
```diff
@@ -710,8 +738,16 @@ static int cmd_seek(void *data, const char *input) {
...
+				break;
+			default:
+				cmd_seek_opcode (core, input + 1);
```
Comment (radare) [inline libr/core/cmd_seek.c]: probably better to just handle ' ' and \0 here, and show error on default

File: libr/util/list.c [hunk 5]
```diff
@@ -607,6 +607,18 @@ R_API RList *r_list_uniq(const RList *list, RListComparator cmp) {
...
+	RListIter *iter;
+	RStrBuf *buf = r_strbuf_new ("");
+	if (!buf) {
+		return NULL;
+	}
+	char *item;
+	r_list_foreach (list, iter, item) {
+		r_strbuf_appendf (buf, "%s%c", item, ch);
+	}
+	return r_strbuf_get (buf);
```
Comment (radare) [inline libr/util/list.c]: use r_strbuf_drain, otherwise its leaking

File: libr/core/cmd_seek.c (hunk 2 above)
Comment (deepakchethan) [reply libr/core/cmd_seek.c]: input+2 should work too. but it is capable of handling full input.

File: libr/core/cmd_seek.c (hunk 3 above)
Comment (deepakchethan) [reply libr/core/cmd_seek.c]: I will handle invalid in default itself

File: libr/core/cmd_seek.c (hunk 4 above)
Comment (deepakchethan) [reply libr/core/cmd_seek.c]: Will do :)

**Comment by deepakchethan** [comment]
> Please add a couple chapters in this section of radare2 book as well https://radare.gitbooks.io/radare2book/content/scripting/intro.html

Will sure do :)

**Comment by XVilka** [comment]
Can you please also add new tests for these commands then?

**Comment by radare** [comment]
There are 7 XX, all of them related to the 's' command

**Comment by radare** [comment]
One xx now:

```
[31m[XX][39m db/cmd/cmd_seek [33mseekop[39m  151

$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-16364z3Y0HF7VnjGt ../bins//mach0/ired-arm64
s
so+3
s
so-3
s
so+2
s
so-2
s

 0x100005864
[35m-0x100005870[39m
[35m-0x100005864[39m
[35m-0x10000586c[39m
[35m-0x100005864[39m
[35m[39m
EXPECT=<<EOF
```

**Comment by deepakchethan** [comment]
> One xx now:
> 
> ```
> �[31m[XX]�[39m db/cmd/cmd_seek �[33mseekop�[39m  151
> 
> $ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-16364z3Y0HF7VnjGt ../bins//mach0/ired-arm64
> s
> so+3
> s
> so-3
> s
> so+2
> s
> so-2
> s
> 
>  0x100005864
> �[35m-0x100005870�[39m
> �[35m-0x100005864�[39m
> �[35m-0x10000586c�[39m
> �[35m-0x100005864�[39m
> �[35m�[39m
> EXPECT=<<EOF
> ```

I think by putting the `cmd_seek_opcode` under default would be better since the command is of the form so+3 with no space

**Comment by radare** [comment]
thanks!

**Pull request merged**